core/third_party_libs/pue/pue-client.php

```diff
@@ -1032,6 +1032,25 @@ private function requestInfo(array $queryArgs = array())
                 if (isset($response->extra_notices)) {
                     $this->add_persistent_notice($response->extra_notices);
                 }
+                // This instance is for EE Core, we have a response from PUE so lets check if it contains PUE Plugin data.
+                if ($this->slug == 'event-espresso-core-reg' && isset($response->extra_data) && !empty($response->extra_data->plugins) ) {
+                    // Pull PUE pugin data from 'extra_data'.
+                    $plugins_array = json_decode($result['body'], true);
+                    $plugins = $plugins_array['extra_data']['plugins'];
```

Comment:
can you not use `$response->extra_data->plugins` here? why decode again?

you've already checked that it exists... or does it not contain the same data that's in raw response?

Reply:
`json_decode($result['body']);` returns stdClass objects.

`json_decode($result['body'], true)` returns assoc arrays.

Soooo, the short answer is I couldn't figure out how to use the stdClass objects without converting them to arrays anyway so just did the lot as I knew how to work with those.

However, I only need `$plugins_array['extra_data']['plugins']` so maybe I should just decode just that with:

`$plugins_array = json_decode($result['body'], true);` 

as `$response->extra_data->plugins` tells me if its empty anyway?

Or am I missing something obvious with the stdClass objects?

Reply:
it looks like you only need to access

```php
$plugins[ $addon->getPueSlug() ]
```

which you should be able to access with a `stdClass` object like this:

```php
$plugins->{ $addon->getPueSlug() }
```

Reply:
Ahhh! Ok... this make sense.

I'll check this out now, thank you!